File: SDKs/googleapis.com/youtube/v3/java/lib/src/main/java/openapisdk/models/shared/ChannelConversionPing.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * ChannelConversionPing
 * Pings that the app shall fire (authenticated by biscotti cookie). Each ping has a context, in which the app must fire the ping, and a url identifying the ping.
**/
public class ChannelConversionPing {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("context")
    public ChannelConversionPingContextEnum context;
    public ChannelConversionPing withContext(ChannelConversionPingContextEnum context) {
        this.context = context;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("conversionUrl")
    public String conversionUrl;
    public ChannelConversionPing withConversionUrl(String conversionUrl) {
        this.conversionUrl = conversionUrl;
        return this;
    }
}

File: SDKs/googleapis.com/youtube/v3/python/src/sdk/models/shared/channelconversionping.py
import dataclasses
from typing import Optional
from enum import Enum
from dataclasses_json import dataclass_json
from sdk import utils

class ChannelConversionPingContextEnum(str, Enum):
    SUBSCRIBE = "subscribe"
    UNSUBSCRIBE = "unsubscribe"
    CVIEW = "cview"


@dataclass_json
@dataclasses.dataclass
class ChannelConversionPing:
    r"""ChannelConversionPing
    Pings that the app shall fire (authenticated by biscotti cookie). Each ping has a context, in which the app must fire the ping, and a url identifying the ping.
    """
    
    context: Optional[ChannelConversionPingContextEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('context') }})
    conversion_url: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('conversionUrl') }})
    
